core/include/sched.h

```diff
@@ -278,6 +278,63 @@ typedef void (*sched_callback_t)(kernel_pid_t active, kernel_pid_t next);
 void sched_register_cb(sched_callback_t callback);
 #endif /* MODULE_SCHED_CB */
 
+#if (IS_USED(MODULE_SCHED_RUNQ_CALLBACK)) || defined(DOXYGEN)
+/**
+ * @brief   Scheduler runqueue callback
+ *          function has to be provided by the user of this API
+ *          it will be called: when the scheduler is run,
+ *                             when a thread enters the active queue or
+ *                             when the last thread leaves a queue
+ *
+ * @param   prio      the priority of the runqueue that changed
+ *
+ * @warning This API is not intended for out of tree users.
+ *          Breaking API changes will be done without notice and
+ *          without deprecation. Consider yourself warned!
+ */
+extern void sched_runq_callback(uint8_t prio);
+#endif
+
+/**
+ * @brief   round robin a runqueue
+ *
+ * @param   prio      the priority of the runqueue to round robin
+ *
+ * @warning This API is not intended for out of tree users.
+ *          Breaking API changes will be done without notice and
+ *          without deprecation. Consider yourself warned!
+ */
+static inline void sched_round_robin_runqueue(uint8_t prio)
+{
+    clist_lpoprpush(&sched_runqueues[prio]);
+}
+
+/**
+ * @brief   count the number of threads in a runqueue up to 2
+ *
+ * @param[in]   prio      the priority of the runqueue to get the length from
+ * @returns     number of threads in a runqueue up to 2
+ * @retval 0    The runqueue is empty
+ * @retval 1    There is exactly one thread in the runqueue
+ * @retval 2    There are at least two threads in the runqueue
+ *
+ * @warning This API is not intended for out of tree users.
+ *          Breaking API changes will be done without notice and
+ *          without deprecation. Consider yourself warned!
+ */
+static inline uint_fast8_t sched_runqueue_len2(uint8_t prio)
+{
+    uint_fast8_t c = 0;
+
+    if (sched_runqueues[prio].next) {
+        ++c;
```

Comment:
unless you look at `/core/include/clist.h: clist_count(  ) `

reason might be: the prefix ++ is more efficient in some cases (i think O0 for that simple statement, for more complex staments it is more effictive for higher Optimizations)

Reply:
yeah, I stumbled on that when looking at how hard it would be to implement `clist_count_upto()`. :)
I immediately git blamed that. Seems like I missed that when reviewing.

I don't think that one way or the other would change the resulting code in this case. I think it is style, and I'd like to use my maintainer pass here.

Reply:
I think that if `++c` indeed yields more efficient code than `c++` in cases the result is not used, you can file a bug for the compiler ;-)

Reply:
k, i change that or implement the clist_is_empty, clist_is_len_1, clist_longer_than_1 function.